src/gt4py/storage/storage.py

```diff
@@ -212,13 +214,17 @@ def __array_finalize__(self, obj):
                     raise RuntimeError(
                         "Meta information can not be inferred when creating Storage views from other classes than Storage."
                     )
-                if self.ndim != obj.ndim and self.ndim != 0:
-                    raise RuntimeError(
-                        "Dimension reducing slicing storages is not supported. Use `Storage.to_numpy()` to retrieve a "
-                        "slicable numpy array and create a new storage from it with proper metadata if neccessary."
-                    )
                 self.__dict__ = {**obj.__dict__, **self.__dict__}
                 self.is_stencil_view = False
+                if hasattr(obj, "_new_index"):
+                    index_iter = itertools.chain(
+                        obj._new_index, [slice(None, None)] * (len(obj.mask) - len(obj._new_index))
+                    )
+                    sliced_indices = (isinstance(x, slice) for x in index_iter)
+                    self._mask = obj.mask and gt_utils.interpolate_mask(
+                        sliced_indices, obj.mask, False
```

Comment:
One question: why is `and` used here? This is **not** doing an element-wise `and` operation between the components of the current mask and the interpolated mask

Reply:
oh, I experimented with `and` and I thought it did do an element-wise `and`. I'll double check the behavior today!